pandas/tests/io/test_fsspec.py

```diff
@@ -0,0 +1,92 @@
+import numpy as np
+import pytest
+
+from pandas import DataFrame, date_range, read_csv, read_parquet
+import pandas._testing as tm
+from pandas.util import _test_decorators as td
+
+df1 = DataFrame(
+    {
+        "int": [1, 3],
+        "float": [2.0, np.nan],
+        "str": ["t", "s"],
+        "dt": date_range("2018-06-18", periods=2),
+    }
+)
+text = df1.to_csv(index=False).encode()
```

Comment:
cc @simonjayhawkins, mypy is complaining here since to_csv can technically return None, but it's not here since we don't specify a path. Can we `type: ignore` this?

Reply:
since the return type of to_csv is Optional[str] and the return value is dependent on the type of path_or_buf rather than the value of path_or_buf, an overload doesn't need Literals, so doing this properly shouldn't be too hard. similar done recently in #34207

to keep this PR focussed, type: ignore is the easiest. A comment for later reference may help.

(could also maybe split the assignment so the ignore is only for the .encode() call.)

or overload as a pre-cursor PR.

Reply:
Oh cool, I didn't know that overload handled this case.

So I think a type: ignore for now, and a followup issue to remove it by overloading the signature.